cmake/kokkos_tribits.cmake

```diff
@@ -280,6 +280,10 @@ FUNCTION(KOKKOS_LINK_TPLS LIBRARY_NAME)
     TARGET_INCLUDE_DIRECTORIES(${LIBRARY_NAME} PUBLIC ${HPX_INCLUDE_DIRS})
   ENDIF()
 
+  IF (KOKKOS_ENABLE_PTHREAD)
+    TARGET_LINK_LIBRARIES(${LIBRARY_NAME} PRIVATE Threads::Threads)
```

Comment:
The link needs to be PUBLIC for the application linking to use the threads library.  I.E.  Change PRIVATE to PUBLIC or this fails most of the spot checks.